fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -1029,29 +956,94 @@ where
     pub(crate) fn run_program(
         &mut self,
     ) -> Result<ProgramState, InterpreterError<S::DataError>> {
-        loop {
-            // Check whether the instruction will be executed in a call context
-            let in_call = !self.frames.is_empty();
-
-            let state = self.execute()?;
-
-            if in_call {
-                // Only reverts should terminate execution from a call context
-                match state {
-                    ExecuteState::Revert(r) => return Ok(ProgramState::Revert(r)),
-                    ExecuteState::DebugEvent(d) => return Ok(ProgramState::RunProgram(d)),
-                    _ => {}
-                }
-            } else {
-                match state {
-                    ExecuteState::Return(r) => return Ok(ProgramState::Return(r)),
-                    ExecuteState::ReturnData(d) => return Ok(ProgramState::ReturnData(d)),
-                    ExecuteState::Revert(r) => return Ok(ProgramState::Revert(r)),
-                    ExecuteState::DebugEvent(d) => return Ok(ProgramState::RunProgram(d)),
-                    ExecuteState::Proceed => {}
+        let Some(script) = self.transaction().as_script() else {
+            unreachable!("Only `Script` transactions can be executed inside of the VM")
+        };
+        let gas_limit = *script.script_gas_limit();
+
+        let (result, state) = if script.script().is_empty() {
+            // Empty script is special-cased to simply return `1` as successful execution.
+            let return_val = 1;
+            self.ret(return_val)?;
+            (
+                ScriptExecutionResult::Success,
+                ProgramState::Return(return_val),
+            )
+        } else {
+            // TODO set tree balance
```

Comment:
I know this is not introduced in this PR so it's not blocking, but do we have an issue for this?

Reply:
Nothing that I know of. I'm not even sure what it's actually about.